File: platforms/hacker-earth/E--make-an-array-85abd7ad/solutions/solution.py
def solve(N, A):
    s = sum(A)
    m = max(A)
    if s % (N - 1) != 0:
        return -1

    k = s // (N - 1)
    if m > k:
        return -1

    return k


T = int(input())
for _ in range(T):
    N = int(input())
    A = list(map(int, input().split()))

    out_ = solve(N, A)
    print(out_)

File: platforms/hacker-earth/E--make-an-array-85abd7ad/solutions/solution.go
package main

import "fmt"

func solve(N int, A []int) int {
	sum := 0
	maxVal := 0

	for i := 0; i < N; i++ {
		sum += A[i]
		if A[i] > maxVal {
			maxVal = A[i]
		}
	}

	if sum%(N-1) != 0 {
		return -1
	}

	k := sum / (N - 1)

	if maxVal > k {
		return -1
	}

	return k
}

func main() {
	var T int
	fmt.Scanln(&T)
	for i := 0; i < T; i++ {

		var N int
		fmt.Scanln(&N)
		A := make([]int, N)
		for i_A := 0; i_A < N; i_A++ {
			fmt.Scan(&A[i_A])
		}

		var out_ int = solve(N, A)
		fmt.Print(out_)
		fmt.Println()
	}
}
